enclave-runtime/src/lib.rs

```diff
@@ -363,11 +367,20 @@ pub unsafe extern "C" fn init_direct_invocation_server(
 	let rpc_responder = Arc::new(RpcResponder::new(connection_registry.clone()));
 
 	let side_chain_api = Arc::new(SideChainApi::<itp_types::Block>::new());
-	let top_pool = BPool::create(PoolOptions::default(), side_chain_api, rpc_responder);
+	let top_pool = Arc::new(BPool::create(PoolOptions::default(), side_chain_api, rpc_responder));
+	let state_facade = Arc::new(StateFacade);
+
+	let rsa_shielding_key = match Rsa3072Seal::unseal() {
+		Ok(k) => k,
+		Err(e) => {
+			error!("Failed to unseal shielding key: {:?}", e);
+			return sgx_status_t::SGX_ERROR_UNEXPECTED
+		},
+	};
 
-	GlobalTopPoolContainer::initialize(top_pool);
+	let rpc_author = Arc::new(Author::new(top_pool, state_facade, rsa_shielding_key));
 
-	let rpc_author = Arc::new(Author::new(Arc::new(GlobalTopPoolContainer)));
+	GlobalAuthorContainer::initialize(rpc_author.clone());
```

Comment:
Instead of initializing a global top pool, we now initialize a global RPC author (which contains the top pool).